src/megaapi_impl.cpp

```diff
@@ -24907,7 +24889,7 @@ void MegaFolderDownloadController::downloadFolderNode(MegaNode *node, string *pa
     }
     else
     {
-        children = megaApi->getChildren(node);
+        children = megaApi->getChildren(node, MegaApi::ORDER_NONE);  // no order is much faster for a very large folder (or nested folders with large subfolders)
```

Comment:
Good catch!!
Maybe we should set `ORDER_NONE` as default value? I can see other usages where the default order is used, but I hesitate it's in purpose, like in:
- `MegaApiImpl::getBackupFolders()`
- `MegaApiImpl::authorizeMegaNodePrivate()`
- `MegaBackupController::removeexceeding()`
All of them can benefit of the `ORDER_NONE`, since the order is irrelevant for them (but maybe not for apps calling `MegaApi::getChildren()`

Reply:
That sounds like a good improvement, I would like to keep it in mind and do it in a separate PR though, on develop, since this PR is heading for a release branch, as it will require some time and testing.

Reply:
That sounds like a good improvement, though I'm not sure if it's right for this PR which is just about speeding up the large transfer case.   I think we should keep it in mind for a future PR, and especially if we do see some performance issues in those other cases.

Reply:
I would add it to this PR. In the end, the PR aims to solve performance issues and you found out that sorting was a bottleneck on large sets. So... why not to improve all over the code now that it has been analyzed? Specially when it's just 3 lines :)

Reply:
Ok, I've gone through them and I agree they are suitable in call cases - changes made.  In fact I removed the default so it's clear what we are doing each time we call it (it's a private interface, and only a few cases used the default anyway)